### PR TITLE
soc: arm: nrf9160: add missing NRF_UICR definition

### PR DESCRIPTION
**Pull request by ioannisg**
NRF_UICR needs to be defined for nRF9160 SoC in
nrfx_config_nrf9160.h, because it is not defined
in nrfx/hal/mdk/nrf9160.h (as it is a Secure-only
peripheral).

Signed-off-by: Ioannis Glaropoulos <Ioannis.Glaropoulos@nordicsemi.no>

### CODE REVIEW

**Comment by ioannisg** [comment]
It's a bug because NRF_UICR is used in the nrf flash driver and needs to be defined for all nrf SOCs that may use that driver.

**Review by anangl — APPROVED** [review]

**Review by carlescufi — APPROVED** [review]

**Comment by ioannisg** [comment]
Do we need to back-port this @anangl ? :)

**Review by ulfalizer — APPROVED** [review]

**Comment by anangl** [comment]
> Do we need to back-port this @anangl ? :)

I think we do. Someone may appear brave enough to enable the `SOC_FLASH_NRF_UICR` option in a build for nRF9160 in 1.14 and encounter the problem. In 1.14 we'll need a dts alias in order to provide the definition of `NRF_UICR` (`DT_INST_*` macros are not available there), but apart from this, the backporting should be pretty straightforward.

File: ext/hal/nordic/nrfx_config_nrf9160.h [hunk 1]
```diff
@@ -193,6 +193,11 @@
         ((NRF_UARTE_Type *)DT_NORDIC_NRF_UARTE_UART_3_BASE_ADDRESS)
 #endif
 
+#ifdef DT_INST_0_NORDIC_NRF_UICR_BASE_ADDRESS
+#define NRF_UICR \
+        ((NRF_UICR_Type *)DT_INST_0_NORDIC_NRF_UICR_BASE_ADDRESS)
```
Comment (MaureenHelm) [inline ext/hal/nordic/nrfx_config_nrf9160.h]: The rest of this file doesn't use DT_INST macros. Should this be `DT_NORDIC_NRF_UICR_0_BASE_ADDRESS` instead?

Comment (ioannisg) [reply ext/hal/nordic/nrfx_config_nrf9160.h]: So, single-instance peripherals do not need a dts "alias" definition, and then DTC will generate a macro of _this_ format, i.e. DT_INST_0_NORD_NRF_xxxx_BASE_ADDRESS.

There are some other single-instance peripherals with the same format in this file.

**Review by MaureenHelm — APPROVED** [review]

**Pull request merged**